tests/playwright-test/plugins.spec.ts

```diff
@@ -0,0 +1,162 @@
+      export default log;
+    `,
+    'test.spec.ts': `
+      import log from './log';
+      const { test } = pwt;
+      test('it works', async ({}) => {
+      });
+    `,
+    'playwright.config.ts': `
+      import { myPlugin } from './plugin.ts';
+      module.exports = {
+        plugins: [
+            myPlugin('a'),
+            myPlugin('b'),
+        ],
+        globalSetup: 'globalSetup.ts',
+        globalTeardown: 'globalTeardown.ts',
+      };
+    `,
+    'globalSetup.ts': `
+      import log from './log';
+      const setup = async () => {
+        await new Promise(r => setTimeout(r, 100));
+        log('globalSetup');
+      }
+      export default setup;
+    `,
+    'globalTeardown.ts': `
+      import log from './log';
+      const teardown = async () => {
+        await new Promise(r => setTimeout(r, 100));
+        log('globalTeardown');
+      }
+      export default teardown;
+    `,
+    'plugin.ts': `
+      import log from './log';
+      export const myPlugin = (name: string) => ({
+        setup: async () => {
+          await new Promise(r => setTimeout(r, 100));
+          log(name, 'setup');
+        },
+        teardown: async () => {
+          await new Promise(r => setTimeout(r, 100));
+          log(name, 'teardown');
+        },
+      });
+    `,
+  });
+  expect(result.exitCode).toBe(0);
+  expect(result.passed).toBe(1);
+  const logLines = await fs.promises.readFile(log, 'utf8');
+  expect(logLines.split('\n')).toEqual([
+    'a setup',
+    'b setup',
+    'globalSetup',
+    'globalTeardown',
+    'b teardown',
+    'a teardown',
+    '',
+  ]);
+});
+
+test('plugins via require', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'test.spec.ts': `
+      const { test } = pwt;
+      test('it works', async ({}) => {
+        expect(process.env.PW_CONFIG_DIR).toContain('plugins-via-require');
+      });
+    `,
+    'playwright.config.ts': `
+      export default { plugins: [ 'plugin.ts' ] };
+    `,
+    'plugin.ts': `
+      export function setup(config, configDir, suite) {
+        process.env.PW_CONFIG_DIR = configDir;
+      };
+    `
+  });
+  expect(result.exitCode).toBe(0);
+  expect(result.passed).toBe(1);
+});
+
+test('fixtures', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'test.spec.ts': `
+      const { test } = pwt;
+      test('it works', async ({ foo }) => {
+        expect(foo).toEqual(42);
+      });
+
+      test('it uses standard fixture', async ({ myBrowserName }) => {
+        expect(myBrowserName).toEqual('chromium');
+      });
+    `,
+    'playwright.config.ts': `
+      import plugin from './plugin.ts';
+      module.exports = {
+        plugins: [ plugin ],
+      };
+    `,
+    'plugin.ts': `
+      export default {
+        fixtures: {
+          foo: 42,
+          myBrowserName: async ({ browserName }, use) => { await use(browserName) }
```

Comment:
We don't need to do this now, but we should make sure the various fixture options work and/or still validate nicely.